pkg/providers/amifamily/resolver.go

```diff
@@ -130,21 +131,32 @@ func (r Resolver) Resolve(ctx context.Context, nodeClass *v1beta1.EC2NodeClass,
 	}
 	var resolvedTemplates []*LaunchTemplate
 	for amiID, instanceTypes := range mappedAMIs {
```

Comment:
Is it worth breaking out some of this logic into a separate function now that the launch template generation logic has gotten more complicated?